### DISCUSSION TITLE
Incorrect Address for pool fees

### DISCUSSION THREAD

**Discussion by davoodi-dre** [coinfoundry/miningcore#1607] [Q&A]
I am not sure if this is an issue with config I have or an issue with the daemon. But for some reason in my mining pool every time a block is hit, and a payout is processed, the pool fee is sent to some random wallet address. The miner is getting their deposit which is good however the pool itself is not making anything as the pool fees are going somewhere random.

Here is my config, any help appreciated.

{
    "logging": {
      "level": "info",
      "enableConsoleLog": true,
      "enableConsoleColors": true,
      "logFile": "pool.log",
      "logBaseDirectory": "/",
      "perPoolLogFile": false
    },
    "banning": {
      "manager": "Integrated",
      "banOnJunkReceive": true,
      "banOnInvalidShares": false
    },
    "notifications": {
      "enabled": true,
      "email": {
        "host": "XXX",
        "port": 587,
        "user": "XXX",
        "password": "XXX",
        "fromAddress": "XXX",
        "fromName": "DGB Mining Pool"
      },
      "admin": {
        "enabled": true,
        "emailAddress": "XXX",
        "notifyBlockFound": false
      }
    },
    "api": {
      "enabled": true,
      "listenAddress": "0.0.0.0",
      "port": 4000,
      "metricsIpWhitelist": [],
      "rateLimiting": {
        "disabled": true,
        "rules": [
          {
            "Endpoint": "*",
            "Period": "1s",
            "Limit": 20
          }
        ],
        "ipWhitelist": []
      }
    },
    "persistence": {
      "postgres": {
        "host": "127.0.0.1",
        "port": 5432,
        "user": "miningcore",
        "password": "XXX",
        "database": "miningcore"
      }
    },
    "paymentProcessing": {
      "enabled": true,
      "interval": 600,
      "shareRecoveryFile": "recovered-shares.txt"
    },
    "pools": [
      {
        "id": "dgb-sha256-1",
        "enabled": true,
        "coin": "digibyte-sha256",
        "address": "Wallet Address",
        "rewardRecipients": [
          {
            "address": "Wallet Address",
            "percentage": 1.0
          }
        ],
        "blockRefreshInterval": 500,
        "jobRebroadcastTimeout": 10,
        "clientConnectionTimeout": 600,
        "banning": {
          "enabled": true,
          "time": 600,
          "invalidPercent": 50,
          "checkThreshold": 50
        },
        "ports": {
          "3062": {
            "listenAddress": "0.0.0.0",
            "difficulty": 512,
            "varDiff": {
              "minDiff": 1024,
              "maxDiff":1024,
              "targetTime": 15,
              "retargetTime": 90,
              "variancePercent": 30
            }
          }
        },
        "daemons": [
          {
            "host": "127.0.0.1",
            "port": 14024,
            "user": "XXX",
            "password": "XXX"
          }
        ],
        "paymentProcessing": {
          "enabled": true,
          "minimumPayment": 0.5,
          "payoutScheme": "PROP",
          "payoutSchemeConfig": {
            "factor": 0.5
          }
        }
      }
    ]
  }

**Comment by oliverw** [comment]
If you set the `rewardRecipients.address` to the pool wallet address then the fee is simply withheld from the payouts. In short, it goes nowhere and accumulates in the pool wallet.

**Reply by davoodi-dre** [reply]
I have done that, and I still have the same issue. Every time it hits a block, it processes the payment to the miner wallet and then for some reason it sends the fee to some other random wallet address.

**Reply by davoodi-dre** [reply]
After further digging. I have found it to be with the daemon and not the mining pool.

**Reply by edcomrocha** [reply]
Could you tell me what the error is?
I'm having the same behavior in some daemons.
Thanks

**Reply by jvxis** [reply]
Digibyte is a bitcoin fork, right? Could you check if creating a legacy address for the pool wallet is possible? I experienced the same issue with bitcoin forks, and it was solved by generating a Legacy address and using it for the pool wallet. Usually, it is a parameter to generate the address, like coin-cli newaddress "label" "legacy". This is my best guess.

**Reply by oliverw** [reply]
> Digibyte is a bitcoin fork, right? Could you check if creating a legacy address for the pool wallet is possible? I experienced the same issue with bitcoin forks, and it was solved by generating a Legacy address and using it for the pool wallet. Usually, it is a parameter to generate the address, like coin-cli newaddress "label" "legacy". This is my best guess.

This could definitely be the culprit. 👍

**Reply by davoodi-dre** [reply]
to get around it I changed my fee pool to an external wallet address but left the pool address the same. It all seems to work well now.

**Reply by ghost** [reply]
This is my problem with novo coin. No matter what address I set. It only sends the first payment to the specified address, but generates a new address for the next payment. If I restart miningcore, it sends it to the specified address the first time, and then generates a new address.

**Reply by ghost** [reply]
I generated a legacy address. I'm testing it now

**Reply by ghost** [reply]
Not work.  The block reward arrives at the address, but when paying "[Bitcoin Payout Handler] sendmany returned error: Account has insufficient funds code -6 "

**Reply by jvxis** [reply]
Is there in the pool wallet enough balance to pay the fees?

**Reply by ghost** [reply]
Yes. It has the blockreward amount on it, and confirmed.

**Reply by jvxis** [reply]
What is your payment factor in miningcore config.json?

Just to test, I think you need to send some funds to this Wallet and see what happens.

**Reply by ghost** [reply]
the payment factor 0.2 
Payment worked fine until I used a legacy address. (Except that you always generated a new address for the pool fee)
 I try will send as many funds as are waiting for payment (plus a little more)

**Comment by fidencio-monroy** [comment]
I have the same issue:

I cloned the current version (Apr 15, 2023) to develop stratum protocol in my software to connect to an instance of this pool and I found in the coinbase transaction, that the address for which it would be mining is not the one specified in config.json file.

Does the reward first go to other address before returing to the pool wallet to make payments? is there a fee for using this software?

This is an example of the coinbase transaction the code generates:

0x01000000010000000000000000000000000000000000000000000000000000000000000000ffffffff1d0312fc0b043cde396400b000000f012345670a4d696e696e67636f726500000000020000000000000000266a24aa21a9edd3f13bdb74a5a97a3d45d7fb3ac61bcb90157e648fda7f230055cfa56351a19c2ae38425000000001976a91450b8d6bc09363c6dde83dec43d7063160d0bc32f88ac00000000

if you decode this raw transaction, here is the output:

{
  "txid": "7c39811e67e34a454d0ebdbbb9b082a5d442c0245fbcc535ec19cffa62d8d23f",
  "hash": "7c39811e67e34a454d0ebdbbb9b082a5d442c0245fbcc535ec19cffa62d8d23f",
  "version": 1,
  "size": 161,
  "vsize": 161,
  "weight": 644,
  "locktime": 0,
  "vin": [
    {
      "coinbase": "0312fc0b043cde396400b000000f012345670a4d696e696e67636f7265",
      "sequence": 0
    }
  ],
  "vout": [
    {
      "value": 0.00000000,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_RETURN aa21a9edd3f13bdb74a5a97a3d45d7fb3ac61bcb90157e648fda7f230055cfa56351a19c",
        "hex": "6a24aa21a9edd3f13bdb74a5a97a3d45d7fb3ac61bcb90157e648fda7f230055cfa56351a19c",
        "type": "nulldata"
      }
    },
    {
      "value": 6.29465898,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 50b8d6bc09363c6dde83dec43d7063160d0bc32f OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "76a91450b8d6bc09363c6dde83dec43d7063160d0bc32f88ac",
        "address": "18MpVAyBVZWg2VNGFYSdygSq1a62jpEC96",
        "type": "pubkeyhash"
      }
    }
  ]
}

the address that here appears ("address": "18MpVAyBVZWg2VNGFYSdygSq1a62jpEC96") is valid but it's not the one I configured for the pools' wallet:

{
  "isvalid": true,
  "address": "18MpVAyBVZWg2VNGFYSdygSq1a62jpEC96",
  "scriptPubKey": "76a91450b8d6bc09363c6dde83dec43d7063160d0bc32f88ac",
  "isscript": false,
  "iswitness": false
}

Any ideas?

Thanks.


FM

**Comment by ghost** [comment]
I experience the same with Fennec(FNNC), but here the block submitted also writes a different address than the wallet address specified in the config.


Pool config:
{
"id": "fennec",
"enabled": true,
"coin": "fennec",
"address": "FgQVBJTCXL3YVTfavAdSiMxH5YDtc7u3Ny",
"rewardRecipients": [
{
"address": "FgQVBJTCXL3YVTfavAdSiMxH5YDtc7u3Ny",
"percentage": 0
}
],

But in the daemon logfile: 

[2023-04-27 11:59:20.3378] [I] [fennec] Broadcasting job 0000009a
[2023-04-27 11:59:20.3416] [I] [fennec] Daemon accepted block 85996 [815c83cccb063c7cd48910e5c42bf42fcbd58edb0bca63324d8fa8824ee53371] submitted by FpysUGvXS64Rtn2cJqdf5Cqbdh6KM7sE3g
[2023-04-27 11:59:20.3416] [I] [fennec] [0HMQ70C9JM7E1] Share accepted: D=5.155
[2023-04-27 11:59:20.3594] [I] [fennec] [0HMQ70C9JM7E1] Share accepted: D=5.155
[2023-04-27 11:59:22.6843] [I] [fennec] [0HMQ70C9JM7E1] Share accepted: D=5.155
[2023-04-27 11:59:28.7144] [I] [fennec] [Bitcoin Payout Handler] Block 85996 classified as orphaned due to daemon error -5
[2023-04-27 11:59:30.3469] [I] [fennec] Broadcasting job 0000009b


The reward is sent to an unknown address and then marks it as "orphan".

Other yescryptR16 algo coins work flawlessly (e.g. goldcash, qogecoin, reacoin)

**Reply by fidencio-monroy** [reply]
I think this is not an error, because the addresses are valid and they need to be carefully encoded.
